### ISSUE TITLE
Ident Paths always empty string of loaded packages

### ISSUE THREAD

**Issue by myshkin5** [dave/dst#50]
I'm trying to load a package and walk the resulting tree. I'm noticing that all of the `dst.Ident` `Path` values are set to an empty string if the type is in the package being loaded but the `Path` values are set correctly if the type refers to another package. I pasted a simple program below that loads the `net/http` package and finds the `CookieJar` interface for which it prints an identity of just `CookieJar` (I expected `net/http.CookieJar`). It also prints the identity of the type of the first parameter to the `SetCookies` method as `net/url.URL`.

Is there some way to have the path info filled in?

```go
package main

import (
	"github.com/dave/dst"
	"github.com/dave/dst/decorator"
	"golang.org/x/tools/go/packages"
)

func main() {
	pkgs, err := decorator.Load(&packages.Config{
		Mode: packages.NeedName |
			packages.NeedFiles |
			packages.NeedCompiledGoFiles |
			packages.NeedImports |
			packages.NeedTypes |
			packages.NeedTypesSizes |
			packages.NeedSyntax |
			packages.NeedTypesInfo,
	}, "net/http")
	if err != nil {
		panic(err)
	}

	for _, pkg := range pkgs {
		for _, file := range pkg.Syntax {
			for _, decl := range file.Decls {
				if genDecl, ok := decl.(*dst.GenDecl); ok {
					for _, spec := range genDecl.Specs {
						if typeSpec, ok := spec.(*dst.TypeSpec); ok {
							if typeSpec.Name.Name == "CookieJar" {
								println("CookieJar ident:", typeSpec.Name.String())
								for _, method := range typeSpec.Type.(*dst.InterfaceType).Methods.List {
									if method.Names[0].Name == "SetCookies" {
										param1Type := method.Type.(*dst.FuncType).Params.List[0].Type.(*dst.StarExpr)
										println("  Param1 StarExpr X ident:", param1Type.X.(*dst.Ident).String())
									}
								}
							}
						}
					}
				}
			}
		}
	}
}
```

Output:

```
CookieJar ident: CookieJar
  Param1 StarExpr X ident: net/url.URL
```

**Comment by dave** [comment]
This is by design, so that if you change the path of a package, the local identifiers don't remain pointing at the old path. Does it cause problems?

**Comment by myshkin5** [comment]
It does make a bit more work for me. To give you the full picture, I'm doing code generation on interfaces roughly following these steps:
1. User targets an interface in a package
1. I call `Load` for the given package
1. If the interface includes a nested interface in another package, I call `Load` on that package too

I now have a mix of `*dst.TypeSpec`s from both packages in a map keyed off of the `dst.Ident` of the interface. I fixed up the identities in the keys so they include the package but when I reach into the type spec values, I currently don't know which package it originally came from. I could assume that whenever `Ident.Path` is an empty string (down within the values of the type spec), the package is actually that of the parent type spec but I'm not always hanging on to the original type spec. I could also walk the type spec and "fill in" the paths but I'd prefer not to have to modify what I'm getting.

This code all worked fine when I didn't support nested interfaces but those were simpler days.

Any suggestions are greatly appreciated. Thank you for providing such a valuable library!

**Comment by dave** [comment]
Yeah I can only suggest walking the tree and filling in all the blank paths. This isn't ideal I know.

**Comment by myshkin5** [comment]
👍

**Comment by myshkin5** [comment]
@dave, On closer inspection, I see that I can accomplish what I need by commenting out these three lines in the decorator:
https://github.com/dave/dst/blob/bce17d0c83acd585b79d2a66855e6024edb0ba97/decorator/decorator.go#L331-L333

If you give me some guidance, I would be glad to submit a PR that somehow makes these lines optional. I'm still getting familiar with your library but it seems like the easiest solution would be to add an option to the decorator. I also see the potential for updating the `Load` function to drive this optional behavior possibly via an `Env` value in the `Config`.

Let me know what you think and thanks again!

**Comment by dave** [comment]
Sure... making this optional sounds good... It was a while ago I last worked on this codebase... not sure where the optional flag would best live... let me know what you think and I'll have a quick look later today. Feel free to send a PR and I'll take a look.

**Comment by myshkin5** [comment]
Reopening with impending PR
